spec/models/task_spec.rb

```diff
@@ -1726,4 +1726,137 @@ def next_assignee(_options = {})
       end
     end
   end
+
+  describe ".conditionally_set_dispatch_as_parent" do
+    let(:user) { create(:user) }
+    let(:appeal) { create(:appeal) }
+    let(:root_task) { RootTask.find_or_create_by!(appeal: appeal, assigned_to: Bva.singleton) }
+    let(:parent_task) { create(:task, appeal: appeal, parent: root_task) }
+    before { FeatureToggle.enable!(:block_at_dispatch) }
+    after { FeatureToggle.disable!(:block_at_dispatch) }
+    subject { FoiaTask.create!(appeal: appeal, parent: parent_task, assigned_to: create(:user)) }
+
+    shared_examples "Parent descendant of BvaDispatchTask" do
+      let(:parent_task) { ColocatedTask.find_by(appeal: appeal) }
+
+      before do
+        ColocatedTask.create!(assigned_by: create(:user),
+                              assigned_to: create(:organization),
+                              parent: dispatch_task,
+                              appeal: dispatch_task.appeal)
+      end
+      it "uses the provided task as the parent" do
+        expect(subject.parent).to eq(parent_task)
+      end
+    end
+
+    shared_examples "organization-assignee BvaDispatchTask as parent" do
+      context "that is active" do
+        before do
+          BvaDispatchTask.find_by(appeal: appeal,
+                                  assigned_to_type: "Organization").update!(status: Constants.TASK_STATUSES.assigned)
+        end
+
+        it "uses that as the parent" do
+          expect(subject.parent).to eq(dispatch_org_task)
+        end
+      end
+
+      context "that is on_hold" do
+        before do
+          BvaDispatchTask.find_by(appeal: appeal,
+                                  assigned_to_type: "Organization").update!(status: Constants.TASK_STATUSES.on_hold)
+        end
+
+        it "uses that as the parent" do
+          expect(subject.parent).to eq(dispatch_org_task)
+        end
+
+        context "but the provided task is a descendant of BvaDispatchTask" do
+          let(:dispatch_task) { dispatch_org_task }
+
+          include_examples "Parent descendant of BvaDispatchTask"
+        end
+      end
+
+      context "that is completed" do
+        before do
+          BvaDispatchTask.find_by(appeal: appeal,
+                                  assigned_to_type: "Organization").update!(status: Constants.TASK_STATUSES.completed)
+        end
+        it "uses provided task as the parent" do
+          expect(subject.parent).to eq(parent_task)
+        end
+      end
+    end
+
+    context "appeal has user assigned BvaDispatchTask" do
```

Comment:
Since you committed my suggestions, let's make the other code changes consistent too:
```suggestion
    context "appeal has user-assignee BvaDispatchTask" do
```